src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -615,7 +615,7 @@ private MethodDesc CheckAndParseSingleMethodModeArguments(CompilerTypeSystemCont
             if (method == null)
                 throw new CommandLineException($"Method '{singleMethodName}' not found in '{singleMethodTypeName}'");
 
-            if (method.HasInstantiation != (singleMethodGenericArgs != null) ||
+            if (method.HasInstantiation != (singleMethodGenericArgs.Length != 0) ||
                 (method.HasInstantiation && (method.Instantiation.Length != singleMethodGenericArgs.Length)))
```

Comment:
This can be simplified in the same way @jkotas suggested in https://github.com/dotnet/runtime/pull/76820#discussion_r991492583